### PR TITLE
[ITSASU-1011] Upversion libraries

### PR DESCRIPTION
**Pull request by aug24**
Also fix tests to match new stricter auth/content requirements.

### CODE REVIEW

File: it/controllers/SignUpControllerISpec.scala [hunk 1]
```diff
@@ -24,8 +24,11 @@ class SignUpControllerISpec extends ComponentSpecBase {
 
       val res = IncomeTaxSubscription.signUp(testNino)
 
+      println(res)
```
Comment (aug24) [inline it/controllers/SignUpControllerISpec.scala]: This is the bit I got stuck on for ages - the jsonBodyAs test throws an exception, which crashes scalatest.  Simply separating the two tests allowed the error to be seen.

File: project/MicroServiceBuild.scala [hunk 2]
```diff
@@ -57,12 +59,14 @@ private object AppDependencies {
       override lazy val test = Seq(
         "uk.gov.hmrc" %% "bootstrap-backend-play-28" % bootstrapBackendVersion % scope,
         "org.scalatest" %% "scalatest" % scalaTestVersion % scope,
+        "org.scalatestplus" %% "mockito-3-12" % "3.2.10.0" % scope,
         "org.pegdown" % "pegdown" % pegdownVersion % scope,
```
Comment (sb-dev) [inline project/MicroServiceBuild.scala]: I don't think `pegdown` is needed.

File: project/MicroServiceBuild.scala [hunk 3]
```diff
@@ -26,19 +26,21 @@ private object AppDependencies {
...
   private val pegdownVersion = "1.6.0"
-  private val mockitoVersion = "3.7.0"
+  private val mockitoVersion = "4.4.0"
 
   private val scalaJVersion = "2.4.2"
 
   private val reactiveMongoVersion = "8.0.0-play-28"
 
-  private val wiremockVersion = "2.21.0"
+  private val wiremockVersion = "2.21.0" //2.32.0
```
Comment (sb-dev) [inline project/MicroServiceBuild.scala]: Did you try using `2.32.0`?

Comment (sb-dev) [reply project/MicroServiceBuild.scala]: You can add `"com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.13.2" % scope` to make it work.

File: project/MicroServiceBuild.scala [hunk 4]
```diff
@@ -75,11 +79,12 @@ private object AppDependencies {
...
+        "com.typesafe.play" %% "play-test" % playVersion % scope,
         "org.scalatestplus.play" %% "scalatestplus-play" % scalaTestPlusVersion % scope,
+        "com.vladsch.flexmark" % "flexmark-all" % "0.64.0" % scope,
         "org.scalaj" %% "scalaj-http" % scalaJVersion % scope,
         "org.mockito" % "mockito-core" % mockitoVersion % scope,
-        "com.github.fge" % "json-schema-validator" % "2.2.6" % scope,
+        "com.github.fge" % "json-schema-validator" % "2.2.14" % scope,
         "com.github.tomakehurst" % "wiremock" % wiremockVersion % scope
       )
     }.test
```
Comment (sb-dev) [inline project/MicroServiceBuild.scala:89]: I don't think the `val overrides` on line 94 are needed anymore.

File: project/MicroServiceBuild.scala [hunk 5]
```diff
@@ -57,12 +59,14 @@ private object AppDependencies {
       override lazy val test = Seq(
         "uk.gov.hmrc" %% "bootstrap-backend-play-28" % bootstrapBackendVersion % scope,
         "org.scalatest" %% "scalatest" % scalaTestVersion % scope,
+        "org.scalatestplus" %% "mockito-3-12" % "3.2.10.0" % scope,
         "org.pegdown" % "pegdown" % pegdownVersion % scope,
-        "com.typesafe.play" %% "play-test" % PlayVersion.current % scope,
+        "com.typesafe.play" %% "play-test" % playVersion % scope,
         "org.scalatestplus.play" %% "scalatestplus-play" % scalaTestPlusVersion % scope,
+        "com.vladsch.flexmark" % "flexmark-all" % "0.64.0" % scope,
```
Comment (sb-dev) [inline project/MicroServiceBuild.scala]: `0.64.0` is not compatible with Java 8 and it might be good to extract the version in a variable.

**Review by sb-dev — APPROVED** [review]

**Pull request merged**